include/sundials/sundials_control.h

```diff
@@ -0,0 +1,248 @@
+/* -----------------------------------------------------------------
+ * Programmer(s): Daniel R. Reynolds @ SMU
+ * -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2023, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * SUNDIALS accuracy-based adaptivity controller class. These
+ * objects estimate step sizes for time integration methods such
+ * that the next step solution should satisfy a desired temporal
+ * accuracy.
+ * ----------------------------------------------------------------*/
+
+#ifndef _SUNDIALS_CONTROLLER_H
+#define _SUNDIALS_CONTROLLER_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sundials/sundials_context.h>
+#include "sundials/sundials_types.h"
+
+#ifdef __cplusplus /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/* -----------------------------------------------------------------
+ * SUNControl types (currently, only "H" is implemented; others
+ * are planned):
+ *    NONE    - empty controller (does nothing)
+ *    H       - controls a single-rate step size
+ *    HQ      - controls a single-rate step size and method order
+ *    MRI_H   - controls two multirate step sizes
+ *    MRI_TOL - controls slow and fast relative tolerances
+ * ----------------------------------------------------------------- */
+
+typedef enum
+{
+  SUNDIALS_CONTROL_NONE,
```

Comment:
Would the `NONE` controller be the same as a fixed step size and order controller? In that case do we need a `NONE` type?

Reply:
The `NONE` type is used when ARKODE is run with a `NULL`-valued `SUNAdaptController` object.  This is what we `arkSetFixedStep` now does (removes the controller, and adjusts the min/max step sizes in the `SUNTimestepHeuristics` object).